apps/portal/src/utils/errors.js

```diff
@@ -9,24 +9,110 @@ export class HumanReadableError extends Error {
+    const translateMessage = (message, number = null) => {
+        if (number) {
+            return t(message, {number});
+        } else {
+            return t(message);
+        }
+    };
+
+    const isSpecialMessage = (message) => {
+        // IMPORTANT: If you add to this list, also add to myfakeFunction below so that the parser will pick it up.
+        // Yes, there's got to be a better way to do this.
+
+        let specialMessages = [
+            'No member exists with this e-mail address. Please sign up first.',
+            'No member exists with this e-mail address.',
+            'This site is invite-only, contact the owner for access.',
+            'Unable to initiate checkout session',
+            'This site is not accepting payments at the moment.',
+            'Too many attempts try again in {{number}} minutes.',
+            'Too many attempts try again in {{number}} hours.',
+            'Too many attempts try again in {{number}} days.',
+            'Too many different sign-in attempts, try again in {{number}} minutes',
+            'Too many different sign-in attempts, try again in {{number}} hours',
+            'Too many different sign-in attempts, try again in {{number}} days',
+            'Failed to send magic link email'
+        ];
+        if (specialMessages.includes(message)) {
+            return true;
         }
-        return defaultMessage;
+        return false;
+    };
+
+    const prepareErrorMessage = (message = null) => {
+        // Check for a number in message, if found, replace the number with {{number}} and return the number.
+        // Assumes there's only one number in the message. 
+        if (!message) { 
+            return {preparedMessage: 'An error occurred', number: null}; 
+        }
+        const number = message.match(/\d+/);
+        if (number) {
+            message = message.replace(number[0], '{{number}}');
+        }
+        return {preparedMessage: message, number: number ? number[0] : null};
+    };
+
+    // main function
+    if (!error && !alreadyTranslatedDefaultMessage) {
+        return t('An error occurred'); 
+    }
+    if (error instanceof HumanReadableError || error.message) {
+        const {preparedMessage, number} = prepareErrorMessage(error.message);
+        if (isSpecialMessage(preparedMessage)) {
+            return translateMessage(preparedMessage, number);
+        } else {
+            return alreadyTranslatedDefaultMessage || translateMessage(error?.message);
+        }
+    } else {
+        // use the default message if there's no error message
+        // if we don't have a default message either, fall back to a generic message plus the actual error.
+        return alreadyTranslatedDefaultMessage || t('An error occurred:') + ' ' + error.toString();
     }
 }
+
+/* This fails linting, because it's only here so that these strings can get picked up for translation */
+// eslint-disable-next-line no-unused-vars
+const myfakeFunction = () => {
```

Comment:
Could you explain this? I'm not familiar with the parser so I'm just curious why this is necessary.

Reply:
That makes two of us. The problem is that the parser seems to only pick up wrapped strings - it doesn't understand incoming variable responses from the API.  There's probably a more elegant way to do this, although I'm not sure what it is... guess I'd better dig into the parser documentation, huh?

Reply:
The parser is pretty limited.  It finds things wrapped with t(' something ').  https://github.com/i18next/i18next-parser?tab=readme-ov-file#caveats 

I've adjusted how I do it, because it removes the need to maintain strings in two places.  I think it's better?